c10/xpu/XPUFunctions.cpp

```diff
@@ -65,7 +75,7 @@ inline void initDevicePoolCallOnce() {
   c10::call_once(init_flag, initGlobalDevicePoolState);
```

Comment:
`enumDevices` should probably be guarded by `call_once` instead. Right now, it would be possible to call `initGlobalDevicePoolState` twice and corrupt the device list.

Reply:
`initGlobalDevicePoolState` is guarded by `call_once`. So `initGlobalDevicePoolState` only can be called once.

Reply:
That's correct for calls via `initDevicePoolCallOnce`. Someone could (by mistake) still call `enumDevices` twice. But it should not be a problem if we take care to only call `initDevicePoolCallOnce`.